pkg/controller/gslb/status.go

```diff
@@ -128,3 +140,31 @@ func (r *ReconcileGslb) getHealthyRecords(gslb *ohmyglbv1beta1.Gslb) (map[string
 
 	return healthyRecords, nil
 }
+
+func (r *ReconcileGslb) updateManagedHostsTotalMetric(gslb *ohmyglbv1beta1.Gslb, serviceHealth map[string]string) error {
```

Comment:
@somaritane there is quite amount of logic in metric functions, what about unit tests ?

Reply:
@ytsarev 
What is our take on unit-tests for internal (unexported) functions, which are not the part of API/contract? 
I usually take pragmatic approach and trying to avoid this, unless the logic gets really complex (which on other hand is the sign, that there’s a need of some refactoring), and usually unit testing for non-public functions/methods is considered unwanted practice.
So I see following options here:
Make controller metrics a separate package, and export the functions, then cover them by tests. In the future, as we get more complex instrumentation added for metrics in different packages, there will be more satellite metric packages involved.
Cover metrics by integration/e2e tests instead of unit tests
Write unit tests for internal package functions, but decide about balance: should we cover all internal functions  with tests, how much complex is complex etc :slightly_smiling_face:

Reply:
Hi, personally I'd like to go with metrics in separate package as you proposed in 1. , to keep separation of contrac,  better testability and maintainibility.

Reply:
@somaritane 
Not testing unexported functions is arguable. Some people say that the fact that it's the power of golang testing package to be able to test internal functions.
Anyhow in our specific use case most of the functions we use inside Reconcile loop are unexported (for a good reason) . Not testing them would affect quality quite dramatically .
In the meantime the approach to testing is something in between of point 2,3 you provided.

If you looks at our existing unit test suite at https://github.com/AbsaOSS/ohmyglb/blob/master/pkg/controller/gslb/gslb_controller_test.go we stick to roughly following pattern:

* Create special condition under test
* Invoke Gslb reconsiliation loop
* Check the result from outside perspective ( e.g. controlled Ingress was modified, Gslb status got changed in expected way etc)

So we usually do not fine-grain unit test this functions ( until it is unavoidable) rather checking overall result of reconciliation loop with conditions associated with the function under test.

So I would suggest to extend above mentioned test suite with this scenario:

* Create Gslb resource(s) with conditions that about to be reflected in metrics
* Invoke reconciliation loop with `reconcileAndUpdateGslb(t, r, req, cl, gslb)`
* Check if associated metrics are getting exposed as expected

This way we can have a confidence in the metric functions we develop meanwhile avoiding reliance exclusively on very heavy e2e tests.

Reply:
@ytsarev I've added the missing tests (had to remove tight dependency on previous fixture state for some other tests, as I had to restore test conditions there). I also renamed metrics in order to comply to metrics linter, provided with prometheus testutils package.